demisto_sdk/commands/lint/lint_manager.py

```diff
@@ -391,20 +394,25 @@ def execute_all_packages(self,
                             return_warning_code += pkg_status["warning_code"]
                     if pkg_status["pack_type"] not in pkgs_type:
                         pkgs_type.append(pkg_status["pack_type"])
+                logger.info('Finished all futures')
                 return return_exit_code, return_warning_code
         except KeyboardInterrupt:
-            print_warning("Stop demisto-sdk lint - Due to 'Ctrl C' signal")
-            try:
-                executor.shutdown(wait=False)
-            except Exception:
-                pass
+            msg = "Stop demisto-sdk lint - Due to 'Ctrl C' signal"
+            print_warning(msg)
+            logger.warning(msg)
+            executor.shutdown(wait=False)  # If keyboard interrupt no need to wait to clean resources
             return 1, 0
         except Exception as e:
-            print_warning(f"Stop demisto-sdk lint - Due to Exception {e}")
+            msg = f"Stop demisto-sdk lint - Due to Exception {e}"
+            print_warning(msg)
+            logger.error(msg)
             try:
-                executor.shutdown(wait=False)
-            except Exception:
-                pass
+                executor.shutdown(wait=True, cancel_futures=True)  # type: ignore[call-arg]
+            except TypeError:
+                logger.info('Using Python under 3.8, we will cancel futures manually.')
```

Comment:
perhaps better [check the python version directly](https://stackoverflow.com/a/41588204), to avoid catching unwanted `TypeError`s that aren't caused by the version